tf_agents/networks/dueling_q_network_test.py

```diff
@@ -0,0 +1,143 @@
+                                                 tf.float32),
+        action_spec=tensor_spec.BoundedTensorSpec([1], tf.int32, 0, 1),
+        conv_layer_params=((16, 3, 2),))
+    q_values, _ = network(states)
+    self.assertAllEqual(q_values.shape.as_list(), [batch_size, num_actions])
+    self.assertEqual(len(network.trainable_variables), 10)
+
+  @test_util.run_in_graph_and_eager_modes()
+  def test_add_preprocessing_layers(self):
+    batch_size = 3
+    num_actions = 2
+    states = (tf.random.uniform([batch_size, 1]),
+              tf.random.uniform([batch_size]))
+    preprocessing_layers = (
+        tf.keras.layers.Dense(4),
+        tf.keras.Sequential([
+            tf.keras.layers.Reshape((1,)),
+            tf.keras.layers.Dense(4)]))
+    network = dueling_q_network.DuelingQNetwork(
+        input_tensor_spec=(
+            tensor_spec.TensorSpec([1], tf.float32),
+            tensor_spec.TensorSpec([], tf.float32)),
+        preprocessing_layers=preprocessing_layers,
+        preprocessing_combiner=tf.keras.layers.Add(),
+        action_spec=tensor_spec.BoundedTensorSpec(
+            [1], tf.int32, 0, num_actions - 1))
+    q_values, _ = network(states)
+    self.assertAllEqual(q_values.shape.as_list(), [batch_size, num_actions])
+    # At least 2 variables each for the preprocessing layers.
+    self.assertGreater(len(network.trainable_variables), 4)
+
+  @test_util.run_in_graph_and_eager_modes()
+  def test_correct_output_shape(self):
+    batch_size = 3
+    num_state_dims = 5
+    num_actions = 2
+    states = tf.random.uniform([batch_size, num_state_dims])
+    network = dueling_q_network.DuelingQNetwork(
+        input_tensor_spec=tensor_spec.TensorSpec([num_state_dims], tf.float32),
+        action_spec=tensor_spec.BoundedTensorSpec([1], tf.int32, 0, 1))
+    q_values, _ = network(states)
+    self.assertAllEqual(q_values.shape.as_list(), [batch_size, num_actions])
+
+  @test_util.run_in_graph_and_eager_modes()
+  def test_network_variables_are_reused(self):
+    batch_size = 3
+    num_state_dims = 5
+    states = tf.ones([batch_size, num_state_dims])
+    next_states = tf.ones([batch_size, num_state_dims])
+    network = dueling_q_network.DuelingQNetwork(
+        input_tensor_spec=tensor_spec.TensorSpec([num_state_dims], tf.float32),
+        action_spec=tensor_spec.BoundedTensorSpec([1], tf.int32, 0, 1))
+    q_values, _ = network(states)
+    next_q_values, _ = network(next_states)
+    self.evaluate(tf.compat.v1.global_variables_initializer())
+    self.assertAllClose(q_values, next_q_values)
+
+  def test_variables_build(self):
+    num_state_dims = 5
+    network = dueling_q_network.DuelingQNetwork(
+        input_tensor_spec=tensor_spec.TensorSpec([num_state_dims], tf.float32),
+        action_spec=tensor_spec.BoundedTensorSpec([1], tf.int32, 0, 1))
+    self.assertFalse(network.built)
+    variables = network.variables
+    self.assertTrue(network.built)
+    self.assertGreater(len(variables), 0)
+
+

```

Comment:
Can you add a test for the values of the network?

Reply:
Can you tell which tests? 
For example, assertIsNotNone or assertDTypeEqual.

Reply:
I meant with actually verifying the values are what are suppose to be.
`self.assertEqual(expected_q_values, q_values)`

Reply:
Done